src/super_gradients/training/models/segmentation_models/ppliteseg.py

```diff
@@ -0,0 +1,383 @@
+                 out_channels: List[int],
+                 upsample_mode,
+                 align_corners: bool):
+        super().__init__()
+        # Make a copy of channels list, to prevent out of scope changes.
+        encoder_channels = encoder_channels.copy()
+        encoder_channels.reverse()
+        in_channels = encoder_channels.pop(0)
+
+        # TODO - assert argument length
+        self.up_stages = nn.ModuleList()
+        for skip_ch, up_factor, out_ch in zip(encoder_channels, up_factors, out_channels):
+            self.up_stages.append(UAFM(
+                in_channels=in_channels,
+                skip_channels=skip_ch,
+                out_channels=out_ch,
+                up_factor=up_factor,
+                upsample_mode=upsample_mode,
+                align_corners=align_corners
+            ))
+            in_channels = out_ch
+
+    def forward(self, feats: List[torch.Tensor]):
+        feats.reverse()
+        x = feats.pop(0)
+        for up_stage, skip in zip(self.up_stages, feats):
+            x = up_stage(x, skip)
+        return x
+
+
+class PPLiteSegBase(SegmentationModule):
+    """
+    The PP_LiteSeg implementation based on PaddlePaddle.
+    The original article refers to "Juncai Peng, Yi Liu, Shiyu Tang, Yuying Hao, Lutao Chu,
+    Guowei Chen, Zewu Wu, Zeyu Chen, Zhiliang Yu, Yuning Du, Qingqing Dang,Baohua Lai,
+    Qiwen Liu, Xiaoguang Hu, Dianhai Yu, Yanjun Ma. PP-LiteSeg: A Superior Real-Time Semantic
+    Segmentation Model. https://arxiv.org/abs/2204.02681".
+    """
+    def __init__(self,
+                 num_classes,
+                 backbone: AbstractSTDCBackbone,
+                 projection_channels_list: List[int],
+                 sppm_inter_channels: int,
+                 sppm_out_channels: int,
+                 sppm_pool_sizes: List[int],
+                 sppm_upsample_mode: Union[UpsampleMode, str],
+                 align_corners: bool,
+                 decoder_up_factors: List[int],
+                 decoder_channels: List[int],
+                 decoder_upsample_mode: Union[UpsampleMode, str],
+                 head_upsample_mode: Union[UpsampleMode, str],
+                 head_mid_channels: int,
+                 dropout: float,
+                 use_aux_heads: bool,
+                 aux_hidden_channels: List[int],
+                 aux_scale_factors: List[int]
+                 ):
+        """
+        :param backbone: Backbone nn.Module should implement the abstract class `AbstractSTDCBackbone`.
+        :param projection_channels_list: channels list to project encoder features before fusing with the decoder
+            stream.
+        :param sppm_inter_channels: num channels in each sppm pooling branch.
+        :param sppm_out_channels: The number of output channels after sppm module.
+        :param sppm_pool_sizes: spatial output sizes of the pooled feature maps.
+        :param sppm_upsample_mode: Upsample mode to original size after pooling.
+        :param :
```

Comment:
please removeredundant lines

Reply:
It was actually placeholders for rest of args doc, added. [c7ab444](https://github.com/Deci-AI/super-gradients/pull/349/commits/c7ab4449a7723dd379d829a9d2ebca068e4d5b0b)